### PR TITLE
Various rust examples and docs changes

### PR DESCRIPTION
**Pull request by thibault-martinez**
- implement `get_routes`, `get_output_raw` and `get_included_block_raw`
- add missing examples
- clean examples and how tos
- get_output how to

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,8 +13,8 @@ categories = [ "cryptography::cryptocurrencies" ]
 
 [dependencies]
 async-trait = { version = "0.1.56", default-features = false }
-bee-api-types = { git = "https://github.com/iotaledger/bee", rev = "ff52b003a2c3845e2987d23487a696af106c053c", default-features = false }
-bee-block = { git = "https://github.com/iotaledger/bee", rev = "ff52b003a2c3845e2987d23487a696af106c053c", default-features = false, features = [ "serde", "dto", "std" ] }
+bee-api-types = { git = "https://github.com/iotaledger/bee", rev = "4b1ed2a12d02314125e4e5e85a113121182285a4", default-features = false }
+bee-block = { git = "https://github.com/iotaledger/bee", rev = "4b1ed2a12d02314125e4e5e85a113121182285a4", default-features = false, features = [ "serde", "dto", "std" ] }
```
Comment (Thoralf-M) [inline Cargo.toml:17]: Wen update to alpha 1?

File: examples/block/custom_parent.rs [hunk 2]
```diff
@@ -18,19 +18,12 @@ async fn main() -> Result<()> {
 
     // Create a client instance
     let client = Client::builder()
-        .with_node(&node_url) // Insert your node URL here
-        .unwrap()
+        .with_node(&node_url)? // Insert your node URL here
         .finish()?;
 
-    let custom_parent = BlockId::from_str("b5634e05a7c665d7f87330a53633f001a5d1d96b346dc98dc225c4d6c204f23b").unwrap();
+    let custom_parent = BlockId::from_str("b5634e05a7c665d7f87330a53633f001a5d1d96b346dc98dc225c4d6c204f23b")?;
```
Comment (Thoralf-M) [inline examples/block/custom_parent.rs:24]: Probably better to use a new tip, so it actually works

Comment (thibault-martinez) [reply examples/block/custom_parent.rs:24]: I just moved this file, it's part of the files I still have to clean so I'll do it in another PR if that's fine

File: Cargo.toml (hunk 1 above)
Comment (thibault-martinez) [reply Cargo.toml:17]: Soon, just need to release `bee-api-types` now

File: examples/outputs/micro_transaction.rs [hunk 3]
```diff
@@ -67,9 +67,9 @@ async fn main() -> Result<()> {
             ))
             // If the receiver does not consume this output, we Unlock after a day to avoid
             // locking our funds forever.
-            .add_unlock_condition(UnlockCondition::Expiration(
-                ExpirationUnlockCondition::new(address, tomorrow).unwrap(),
-            ))
+            .add_unlock_condition(UnlockCondition::Expiration(ExpirationUnlockCondition::new(
+                address, tomorrow,
```
Comment (Thoralf-M) [inline examples/outputs/micro_transaction.rs]: Not your change, but we should probably at least send from address index 0 to 1, so the address in Expiration and StorageDepositReturn is not the same as in the AddressUnlockCondition

Comment (Thoralf-M) [reply examples/outputs/micro_transaction.rs]: Maybe just do it the same way as in examples/outputs/native_tokens.rs so it's aligned

File: examples/tagged_data_to_utf8.rs [hunk 4]
```diff
@@ -9,10 +9,10 @@ use iota_client::{block::payload::TaggedDataPayload, Client, Result};
 
 #[tokio::main]
 async fn main() -> Result<()> {
-    let tag = hex::decode("68656c6c6f").unwrap();
-    let data = hex::decode("776f726c64").unwrap();
+    let tag = hex::decode("68656c6c6f")?;
+    let data = hex::decode("776f726c64")?;
```
Comment (Thoralf-M) [inline examples/tagged_data_to_utf8.rs]: Maybe add a comment what this is ( `hello` and `world` hex encoded)

**Review by Thoralf-M — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**